### PR TITLE
Tag commits through GraphQL on Github

### LINKED ISSUE #36: Use Github GraphQL to tag a specific commit sha

### PR DESCRIPTION
**Pull request by oanatmaria**
Resolves https://github.com/aserto-dev/scc-lib/issues/36

### CODE REVIEW

File: internal/interactions/gitlabintr.go [hunk 1]
```diff
@@ -17,14 +17,14 @@ type GitlabIntr interface {
 	ListGroups(opt *gitlab.ListGroupsOptions) ([]*gitlab.Group, *gitlab.Response, error)
 	GetProject(pid interface{}) (*gitlab.Project, *gitlab.Response, error)
 	GetNamespace(id interface{}) (*gitlab.Namespace, error)
-	CreateProject(opt *gitlab.CreateProjectOptions) error
+	CreateProject(opt *gitlab.CreateProjectOptions) (*gitlab.Project, error)
```
Comment (carabasdaniel) [inline internal/interactions/gitlabintr.go:20]: As the interface changes here this looks like a backward incompatible change.

File: sources/github.go [hunk 2]
```diff
@@ -433,49 +432,47 @@ func (g *githubSource) InitialTag(ctx context.Context, accessToken *AccessToken,
...
+		tags, err := githubClient.ListRepoTags(ctx, owner, name, &github.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to list tags for repo '%s/%s'", owner, name)
+		}
 
-	if len(tags) > 0 {
-		return nil
-	}
+		if len(tags) > 0 {
+			return nil
```
Comment (carabasdaniel) [inline sources/github.go:449]: This func basically only tags the repository with default tag if the repo doesn't have any tags at all. 

Shouldn't this be actually a fallback behavior ? Using this as a package I would expect to be able to specify the tag I want to create on the repository. 

From the comment I understand the intent was to have this as an initial tag creation mechanism. Would it make sense to change that now ?

File: internal/interactions/gitlabintr.go (hunk 1 above)
Comment (oanatmaria) [reply internal/interactions/gitlabintr.go:20]: this one is in internal and should't impact any user

File: sources/github.go (hunk 2 above)
Comment (oanatmaria) [reply sources/github.go:449]: I wanted to make this function a Tag one but changed my mind along the way as it should stay InitialTag not Tag. I've changed the name back

**Review by carabasdaniel — APPROVED** [review]

**Pull request merged**